packages/core/src/modules/oob/OutOfBandModule.ts

```diff
@@ -464,20 +466,18 @@ export class OutOfBandModule {
         throw new AriesFrameworkError('Dids are not currently supported in out-of-band message services attribute.')
       }
 
+      let existingConnection
       for (const recipientKey of service.recipientKeys) {
-        let existingConnection = await this.connectionsModule.findByTheirKey(recipientKey)
+        const theirDidRecords = await this.dids.findMultipleByVerkey(recipientKey)
 
-        if (!existingConnection) {
+        for (const theirDidRecord of theirDidRecords) {
           // TODO Encode the key and endpoint of the service block in a Peer DID numalgo 2 and using that DID instead of a service block
-          const theirDidRecord = await this.dids.findByVerkey(recipientKey)
-
-          if (theirDidRecord) {
-            existingConnection = await this.connectionsModule.findByDid(theirDidRecord.id)
-          }
+          existingConnection = await this.connectionsModule.findByDid(theirDidRecord.id)
+          // TODO what if we have more connections?
+          if (existingConnection) return existingConnection
         }
-
-        return existingConnection
       }
+      return existingConnection
```

Comment:
We shouldn't find existing connections based on the recipientKeys. We should either transform the service into a peer did method 2, or use the did string.

Not sure we should add this logic as it's not conformant to the RFC

Reply:
Agree, as we discussed in a call we should use peer did method 2. I'll do it in a separate PR.